test/sharness/t0260-sharding.sh

```diff
@@ -137,4 +155,51 @@ test_list_incomplete_dir() {
 
 test_list_incomplete_dir
 
+# Test automatic sharding and unsharding
```

Comment:
As discussed informally in chat (but can elaborate more here if needed) any kind of directory manipulation goes through the same "automatic" sharding mechanism, whether it is adding/removing one entry at a time from an MFS tree with `ipfs files add/rm`, or consuming an entire directory at once with `ipfs add -r <big-dir>`.

So above and below this line we are testing the same UnixFS code. This doesn't mean that the current division is superfluous as we're accessing that mechanism through different `go-ipfs` paths but I'm mentioning this to (a) check if this was indeed the intention here and (b) think if we could re-label this boundary.

Reply:
True, but in the event the code paths were different we'd be testing different things. i.e. here we are testing that when we remove data from a sharded directory we get an unsharded one and when we add the data back we end up with a sharded one again.

Does that make sense?

Reply:
Yes, it does. I think my main issue here is with the "automatic" boundary, which might imply the previous commands were not, that they didn't use the new UnixFS mechanism and instead were forced or explicitly stated to shard (where in fact we calculated the exact entries number that would go over the threshold and trigger the mechanism). I think your previous explanation is more clear (but still we're in the nit realm so we can ignore this):
> here we are testing that when we remove data from a sharded directory we get an unsharded one and when we add the data back we end up with [the same] sharded one again